### PR TITLE
Feature: uploading/view signature 

### PR DESCRIPTION
**Pull request by ziweiwu**
Feature
- allow uploading a signature image file on create/update user page
- add a view button to view the signature image for the react home page and thymeleaf home page
- after click view button, a view signature page will display to show user signature 

Notes
- set function for signature field takes an input file uploaded in the form, convert it to a byte stream and store it in the database 
- `getEncodedSignature()` decode the byte[] signature data from the database into a string using Base64 encoding, it allows the string to be embedded in the HTML and rendered as a picture by browser

### CODE REVIEW

File: src/main/java/github/paz/awardportal/model/User/User.java [hunk 1]
```diff
@@ -85,4 +89,51 @@ public void updateUser(BaseUser update){
...
+    public void setSignature(MultipartFile file) {
+        try {
+            this.signature = file.getBytes();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    // return the signature in base64 string, allow it to be displayed in html
+    public String getEncodedSignature() {
```
Comment (anderMatt) [inline src/main/java/github/paz/awardportal/model/User/User.java:119]: Looks good. However, what do you think about this: https://www.baeldung.com/spring-boot-jsoncomponent

We can customize how User objects are serialized to JSON such that the signature field, if present, is automatically converted to a Base64 string. That way, the front end just needs to access this field .

File: src/main/resources/templates/users/view-signature.html [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    <!-- Bootstrap CSS -->
+    <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.2.1/css/bootstrap.min.css"
+          integrity="sha384-GJzZqFGwb1QTTN6wy59ffF1BuGJpLSa9DkKMp0DgiMDm4iYMj70gZWKYbI706tWS" crossorigin="anonymous">
+
+    <title>View Signature </title>
+</head>
+
+<body>
+<p th:text="'User: ' +  ${user.firstName} + ' '+ ${user.lastName}"></p>
+<img th:src="@{'data:image/jpg;base64,'+${user.getEncodedSignature()}}"/>
```
Comment (anderMatt) [inline src/main/resources/templates/users/view-signature.html:18]: See comment above - this works just fine if we want to mix Thymeleaf templates and our SPA frontend.

File: src/main/java/github/paz/awardportal/model/User/User.java [hunk 3]
```diff
@@ -85,4 +89,51 @@ public void updateUser(BaseUser update){
...
+        try {
```
Comment (anderMatt) [inline src/main/java/github/paz/awardportal/model/User/User.java:120]: Java 8 introduced standard library support for this, which could simplify the code.

String encodedSignature = Base64.getEncoder().encodeToString(signature);

Just pointing this out, this way works perfectly fine too.

Comment (anderMatt) [reply src/main/java/github/paz/awardportal/model/User/User.java:120]: import java.util.Base64

is the import statement

**Review by anderMatt — APPROVED** [review]
Looks good. I had two suggestions, up to you if you want to try them out or keep everything as is:

* Customize User serialization to JSON so that the byte[] signature field is automatically converted to a Base64 string

* Use the standard library java.util.Base64 class for converting the signature byte[] field to an encoded string

File: src/main/java/github/paz/awardportal/model/User/User.java (hunk 1 above)
Comment (ziweiwu) [reply src/main/java/github/paz/awardportal/model/User/User.java:119]: Looks like JSON outputs encoded string by default in `encodedSignature` field.

**Pull request merged**